visual_behavior/data_access/from_lims.py

```diff
@@ -2211,3 +2212,59 @@ def load_objectlist(ophys_experiment_id):
     objectlist_dataframe = pd.read_csv(filepath)
     objectlist_dataframe = lims_utils.update_objectlist_column_labels(objectlist_dataframe)
     return objectlist_dataframe
+
+
+def get_id_type(input_id):
+    '''
+    a function to get the id type for a given input ID
+
+    Examples:
+
+    >> get_id_type(914580664)
+    'ophys_experiment_id'
+
+    >> get_id_type(914211263)
+    'behavior_session_id'
+
+    >> get_id_type(1086515263)
+    'cell_specimen_id'
+
+    >> get_id_type(914161594)
+    'ophys_session_id'
+
+    >> get_id_type(1080784881)
+    'cell_roi_id'
+
+    >> get_id_type(1234)
+    'unknown_id'
+
+    Parameters:
+    -----------
+    input_id : int
+        id to search
+
+    Returns:
+    --------
+    string
+        The ID type, or 'unknown_id' if the ID type cannot be determined
+    '''
+
+    # set up a general query string
+    query = 'select * from {} where id = {} limit 1'
+
+    # iterate over tables
+    for table in ['ophys_experiments', 'ophys_sessions', 'behavior_sessions', 'cell_rois']:
+
+        # check to see if the id is an index in the table
+        if len(db.lims_query(query.format(table, input_id))) > 0:
+            # return if so
+            return table[:-1] + '_id'
```

Comment:
what happens if the id is found multiple times?

Can the same number exist as a ophys_session_id, and a behavior_session_id? Are they assured to be different?

Reply:
I'm fairly certain they are unique. But it's a good point. I'll adjust logic slightly to throw an assertion error if the same ID is found in two places. I'll also iterate through some IDs in some datasets to make sure I can't trigger this assertion.

Reply:
I added an assertion that will throw an error if the ID is found in multiple tables.

I just verified that for a random sample of 200 experiments, the corresponding experiment, session and behavior_session IDs were found in only one table:

```python
import allensdk.brain_observatory.behavior.behavior_project_cache as bpc
from visual_behavior.data_access import loading, from_lims
data_storage_directory = "~\temp" # Note: this path must exist on your local drive
cache = bpc.VisualBehaviorOphysProjectCache.from_s3_cache(cache_dir=data_storage_directory)

experiment_table = cache.get_ophys_experiment_table().reset_index()

for idx, row in experiment_table.sample(200).iterrows():
    for id_type in ['ophys_experiment_id','ophys_session_id','behavior_session_id']:
        
        from_lims.get_id_type(row[id_type])
```

Then I also loaded the cell_specimen_table for all experiments and checked for uniqueness of the cell_specimen and cell_roi IDs for 200 cells:

```python
cell_table = loading.get_cell_table()
for idx, row in cell_table.sample(200).iterrows():
    for id_type in ['cell_roi_id','cell_specimen_id']:
        
        from_lims.get_id_type(row[id_type])
```

Those 200 item random samples aren't definitive, of course. Do you think more extensive verification is necessary? I'll note again that the assertion statement will prevent us from unknowingly being handed an ID from the wrong table if there did happen to be an ID that existed in multiple tables.

Reply:
I think the assertion error is good enough for me